### PR TITLE
Added VariantDataset.collect

### PR DESCRIPTION
**Pull request by tpoterba**
I want this for interface purposes, but it's really not usable due to performance. It takes 2 minutes to collect sample.vcf.

I think that py4j is the main culprit, but the history stuff also slows it down a ton.

### CODE REVIEW

**Comment by jigold** [comment]
How does the history slow it down? It's just a wrapper.

**Comment by tpoterba** [comment]
All the args for each genotype and struct are converted to strings with repr and stored. For certain things like genotype there's no reason to have it extend history_mixin, since we can produce a sensible `repr` without recording the args.

**Comment by tpoterba** [comment]
wait, nevermind. the history stuff was never run because we were creating them with _from_java.

But this means the history is totally broken for genotype + friends, another reason we really should just use repr.

**Comment by tpoterba** [comment]
so it's really just py4j. Man, that stuff is so bad.

**Comment by tpoterba** [comment]
Oh wow, totally forgot. Will do!

**Comment by tpoterba** [comment]
Will also add `take`

**Comment by tpoterba** [comment]
back to you!

File: src/test/scala/is/hail/variant/IntervalSuite.scala [hunk 1]
```diff
@@ -152,7 +152,7 @@ class IntervalSuite extends SparkSuite {
   }
 
   @Test def testFilter() {
-    val vds = hc.importVCF("src/test/resources/sample2.vcf", nPartitions = Some(4)).cache()
+    val vds = hc.importVCF("src/test/resources/sample2.vcf", nPartitions = Some(4)).persist()
```
Comment (jigold) [inline src/test/scala/is/hail/variant/IntervalSuite.scala]: Is this to fix the slow test?

Comment (tpoterba) [reply src/test/scala/is/hail/variant/IntervalSuite.scala]: Oh, oops. I didn't mean to leave this in.

Comment (tpoterba) [reply src/test/scala/is/hail/variant/IntervalSuite.scala]: changing back now.

Comment (tpoterba) [reply src/test/scala/is/hail/variant/IntervalSuite.scala]: done!

File: python/hail/dataset.py [hunk 2]
```diff
@@ -4823,6 +4823,52 @@ def storage_level(self):
         return self._jvds.storageLevel()
 
     @handle_py4j
+    def collect(self):
+        """Collect the dataset into a local list.
+
```
Comment (jigold) [inline python/hail/dataset.py]: Add a warning similar to KeyTable.collect:

```
**Notes**

        This method should be used on very small tables and as a last resort.
        It is very slow to convert distributed Java objects to Python
        (especially serially), and the resulting list may be too large
        to fit in memory on one machine.
```

File: python/hail/dataset.py [hunk 3]
```diff
@@ -4823,6 +4823,52 @@ def storage_level(self):
...
+        **Notes**
+
+        The schema of the returned structs is:
+
+         - **v**: :py:meth:`.VariantDataset.rowkey_schema`
+         - **va**: :py:meth:`.VariantDataset.variant_schema`
+         - **gs**: *list of* :py:meth:`.VariantDataset.genotype_schema`
+
+        :rtype: list of :py:class:`.hail.representation.Struct`
```
Comment (jigold) [inline python/hail/dataset.py]: Missing parameter definition for n.

**Comment by tpoterba** [comment]
did both

**Pull request merged**